server/src/main/java/io/crate/statistics/TransportAnalyzeAction.java

```diff
@@ -67,7 +69,7 @@ public final class TransportAnalyzeAction {
     private final TransportService transportService;
     private final Schemas schemas;
     private final ClusterService clusterService;
-    private final ConcurrentHashMap<FetchSampleRequest, CompletableFuture<Samples>> analysisByRequest = new ConcurrentHashMap<>();
+    private final AtomicReference<CompletableFuture<Samples>> currentRun = new AtomicReference<>();
     private final Executor executor;
 
```

Comment:
Current PR is wrong, currentRun must be used in `fetchSamplesThenGenerateAndPublishStats` to completely reflect state of automatic ANALYZE and behave like "we can have only 1 active ANALYZE"